src/sections/mod.rs

```diff
@@ -242,4 +245,42 @@ impl<'a> PsdCursor<'a> {
 
         Ok(i32::from_be_bytes(array))
     }
+
+    /// Read 'Unicode string' using specified length
+    pub fn read_unicode_string_len(&mut self, length: u32) -> Result<String, Error> {
```

Comment:
This one isn't important so feel free to ignore. I was mainly commenting on `len` functions typically returning `usize` in the standard library so trying to avoid using `len` as a suffix.